docs/content/best-practices/github-workflow.md

```diff
@@ -0,0 +1,162 @@
+````yaml
+run: porter mixins install az --version v0.4.2
+````
+
+### Running Porter commands
+The final part of the workflow is running porter commands. The commands we suggest running are porter install, porter upgrade, porter uninstall, and porter publish. Porter install will install your bundle and give an error if something is wrong. This will be useful as part of your pipeline in testing your bundle because you can go fix the problem right away instead of finding it later. Porter upgrade and porter uninstall will execute the code and also give errors if there are problems. If all these commands run successfully, you can run porter publish to publish your working bundle image to Docker Hub. 
+
+## Setting up the Workflow
+Now that we know the parts that are needed in a workflow, we can learn how to set one up. Here are the steps we will go through to set up the workflow: 
+
+* [Making yaml files](#making-yaml-files)
+* [Setting up secrets in your repository](#setting-up-secrets-in-your-repository)
+* [Using credential files](#using-credential-files)
+
+### Making yaml files
+The way you set up your yaml files depends on who will be contributing to your repository. If you are the only one working on a project and no one else can make pull requests (you have a private repository), you can set up one yaml file to run when a pull request opens and one yaml file to run when a commit is merged. You can have the one that runs on a pull request run your bundle to test it, and you can have the second yaml file publish the bundle to Docker Hub. 
+
+If you are not the only one contributing to the repository and other contributors will be making pull requests from forks, the yaml file setup is slightly different. GitHub actions and workflows do not run automatically on pull requests from a fork. So, in this situation, you will only need one yaml file that runs when a pull request is merged. You can test the bundle and publish in the same yaml file.  
+
+### Setting up secrets in your repository
+In order to publish your bundle to Docker Hub, you will need to set up secrets. You can add these in the settings of your repository and then use them in your workflow.
+
+### Using credential files
+If you were using credentials in your bundle, you will need to set up a credential file in your repository to use with your workflow. After you run `porter credentials generate`, you should see a JSON file with the name of your bundle in the ~/.porter/credentials directory. You need to add this file to your repository so you can pass in the file as the credential. For example, if my bundle was named hello, there would be a file with the credentials called hello.json. I would do the following to install my bundle:
+```yaml
+porter install -c ./hello.json
+```
+
+## Example Code
+
+Now, we will show example code for a workflow and then explain what the code does.
+
+* [Show example code](#show-example-code)
```

Comment:
I think we can omit the links here and the section headers. for "show" and "explain" the example code